rust/src/base.rs

```diff
@@ -81,18 +101,11 @@ impl Base {
         id: BaseId,
         geoid: GeoidString,
         road_network: HaversineRoadNetwork,
-        station_id: Option<StationId>,
         stall_count: usize,
+        station_id: Option<StationId>,
         membership: Option<Membership>,
     ) -> Self {
-        Base {
-            id: Arc::new(id),
-            position: Arc::new(road_network.position_from_geoid(geoid)),
-            membership: Arc::new(membership.unwrap_or_default()),
-            total_stalls: Arc::new(stall_count),
-            available_stalls: Arc::new(stall_count),
-            station_id: Arc::new(station_id),
-        }
+        Base::new(id, geoid, road_network, station_id, stall_count, membership)
```

Comment:
lemme see if i understand this. did we write a custom constructor that does the Arc<> wrapping? that seems like the correct amount of encapsulation, so that building a new Base doesn't require managing the creation of the copies of the fields we are setting.

Reply:
yeah exactly